tests/test_estimators.py

```diff
@@ -69,6 +69,23 @@ def test_estimators_support_continuous_multioutput(estimator, moscow_euclidean):
     estimator.predict(moscow_euclidean.X)
 
 
+@pytest.mark.parametrize("estimator", get_kneighbor_estimator_classes())
+def test_estimators_support_dataframe_indexes(estimator, moscow_euclidean):
+    """All estimators should store and return dataframe indexes."""
+    estimator = estimator(n_neighbors=1)
+    X_df = pd.DataFrame(moscow_euclidean.X, index=moscow_euclidean.ids)
+
+    estimator.fit(moscow_euclidean.X, moscow_euclidean.y)
+    with pytest.raises(NotFittedError, match="fitted with a dataframe"):
+        estimator.kneighbors(X_df, return_distance=False, return_dataframe_index=True)
+
+    estimator.fit(X_df, moscow_euclidean.y)
+    assert_array_equal(estimator.dataframe_index_in_, moscow_euclidean.ids)
+
+    idx = estimator.kneighbors(X_df, return_distance=False, return_dataframe_index=True)
+    assert_array_equal(idx.ravel(), moscow_euclidean.ids)
```

Comment:
Good point! I keep using `moscow_euclidean` just as a convenient way to get some raw data, but you're right that it's potentially confusing since I'm not actually using the Euclidean-specific results.

I think once we wrap up #3, we can replace those uses where we don't care about the actual output with a call to `load_moscow_stjoes`, but for now I think adding an explanatory comment is a good idea!

Reply:
All references to `moscow_euclidean` are gone now, but I think the `k=1` comment is still helpful to clarify why the neighbor IDs match the dataset IDs.